### ISSUE TITLE
Disabling Directory Cache

### ISSUE THREAD

**Issue by foodstuff** [leafo/scssphp#563]
I'm working on a site where I'd like scssphp to use my import search function every time it needs to search for an import.

To clarify, here's my setup;
```
scssroot\default\first.scss
scssroot\default\second.scss
scssroot\siteB\second.scss
```

I'm trying to switch which .scss file is imported based on which domain is being served. In this example I'm getting files from default but if I'm serving `siteB`, I'd like to import from the `siteB` folder.

Currently, if I'm serving `siteA`, scssphp imports files from the `scssroot/default` folder - this is the expected behaviour.
However, if I'm serving `siteB`, scssphp doesn't import the file from the `scssroot/siteB` folder, instead it returns the file from `scssroot/default`.

This runs counter to the behaviour I'm expecting; I've set a callback via `setImportPaths()` that performs searches from the `scssroot` folder and it isn't using it for every search.

scssphp clearly caches the locations of relative folders for performance reasons and that's fine for most purposes - however I'm that very specific use case where it's a problem.

In summary, I'd like to request a configuration option to disable the caching of relative folders and force scssphp to search/use its import paths for every import statement.

(That, or at least some advice on disabling this feature - I've attempted to extend the Compiler class but every change I make breaks the library!)

**Comment by robocoder** [comment]
You can subclass the `Compiler` class and implement your own `findImport` and/or `loadImportFile` methods to achieve what you want.

Since ScssPhp caches files using the realpath (i.e., an absolute path), it sounds like either siteB either sets the import paths in the wrong search order, or never gets called for siteB.

**Comment by robocoder** [comment]
If you're using the example Server class, then no, it's not aware of changes to `setVariables() `or `setImportPaths()` that could affect cached output. This is a known limitation. See #333.
